src/components/timeline/ActiveTabGlobalTrack.js

```diff
@@ -0,0 +1,165 @@
+import type { ConnectedProps } from '../../utils/connect';
+
+type OwnProps = {|
+  +trackReference: GlobalTrackReference,
+  +trackIndex: TrackIndex,
+  +style?: Object /* This is used by Reorderable */,
+  +setInitialSelected: (el: InitialSelectedTrackReference) => void,
+|};
+
+type StateProps = {|
+  +globalTrack: ActiveTabGlobalTrack,
+  +isSelected: boolean,
+  +selectedTab: TabSlug,
+|};
+
+type DispatchProps = {|
+  +selectActiveTabTrack: typeof selectActiveTabTrack,
+|};
+
+type Props = ConnectedProps<OwnProps, StateProps, DispatchProps>;
+
+class ActiveTabGlobalTrackComponent extends PureComponent<Props> {
+  _container: HTMLElement | null = null;
+  _isInitialSelectedPane: boolean | null = null;
+  _selectCurrentTrack = () => {
+    const { selectActiveTabTrack, trackReference } = this.props;
+    selectActiveTabTrack(trackReference);
+  };
+
+  renderTrack() {
+    const { globalTrack } = this.props;
+    switch (globalTrack.type) {
+      case 'tab': {
+        const { threadIndex } = globalTrack;
+        return (
+          <TimelineTrackThread
+            threadIndex={threadIndex}
+            showMemoryMarkers={false}
+          />
+        );
+      }
+      case 'screenshots': {
+        const { threadIndex, id } = globalTrack;
+        return (
+          <TimelineTrackScreenshots threadIndex={threadIndex} windowId={id} />
+        );
+      }
+      default:
+        console.error(
+          'Unhandled active tab globalTrack type',
+          (globalTrack: empty)
+        );
+        return null;
+    }
+  }
+
+  _takeContainerRef = (el: HTMLElement | null) => {
+    const { isSelected } = this.props;
+    this._container = el;
+
+    if (isSelected) {
+      this.setIsInitialSelectedPane(true);
+    }
+  };
+
+  setIsInitialSelectedPane = (value: boolean) => {
+    this._isInitialSelectedPane = value;
+  };
+
+  componentDidMount() {
+    if (this._isInitialSelectedPane && this._container !== null) {
+      this.props.setInitialSelected(this._container);
+    }
+  }
+
+  render() {
+    const { isSelected, style } = this.props;
+
+    return (
+      <li ref={this._takeContainerRef} className="timelineTrack" style={style}>
+        <div
+          className={classNames(
+            'timelineTrackRow timelineTrackGlobalRow activeTab',
+            {
+              selected: isSelected,
+            }
+          )}
+          onClick={this._selectCurrentTrack}
+        >
+          <div className="timelineTrackTrack">{this.renderTrack()}</div>
+        </div>
+        {/* TODO: Render the resources panel here */}
+      </li>
+    );
+  }
+}
+
+export default explicitConnect<OwnProps, StateProps, DispatchProps>({
+  mapStateToProps: (state, { trackIndex }) => {
+    const globalTracks = getActiveTabGlobalTracks(state);
+    const globalTrack = globalTracks[trackIndex];
+    const selectedTab = getSelectedTab(state);
+
+    // These get assigned based on the track type.
+    let isSelected = false;
```

Comment:
Optional: Can this be a selector? It'll re-run this logic every time the component is rendered. I believe this code was copied/pasted.